backend/src/main/java/de/grimsi/gameyfin/rest/LibraryController.java

```diff
@@ -32,20 +29,23 @@ public class LibraryController {
     private final LibraryService libraryService;
     private final ImageService imageService;
 
-    @GetMapping(value = "/scan", produces = MediaType.APPLICATION_JSON_VALUE)
-    public LibraryScanResultDto scanLibrary(@RequestParam(value = "download_images", defaultValue = "true") boolean downloadImages) {
+    @PostMapping(value = "/scan", produces = MediaType.APPLICATION_JSON_VALUE)
```

Comment:
Any reason for the change from GET to POST? I chose GET because POST is usually linked to creating a new resource in REST, which we're not doing here (we're just triggering an action).

Reply:
Yes because POST can contain more data than GET (browsers have different limitations for length [see here](https://www.sistrix.com/ask-sistrix/technical-seo/site-structure/url-length-how-long-can-a-url-be)).
Since the endpoint now allows the client to supply a path for the library to scan which _could_ exceed this length limit.

Of course it would be ridiculous to add such a long path as library, but working as a software dev for almost a decade, I'm not surprised anymore at how creative users are to break applications somehow...

Also technically the scan does create data, so in terms of REST design isn't it more accurate to use POST here?
To my knowledge GET should only be used to retrieve data, especially because browsers tend to cache GET requests which could introduce sideeffects.

Reply:
OK I get your point ✔️